### PR TITLE
[import] backport of some zignature commits from radare2

### PR DESCRIPTION
**Pull request by wargio**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Backports the following commits:

- https://github.com/radareorg/radare2/commit/0a10c3f13a5fb5548d7035ef5e54d2f840424069
- https://github.com/radareorg/radare2/commit/98150ad1829df99e52a96f2ed91fe4399c1ead5c
- https://github.com/radareorg/radare2/commit/086038fecaada0a6db3c8e1c49927614204d675a
- https://github.com/radareorg/radare2/commit/74b8b3cc99f293f101a616099a728247256a8e65
- https://github.com/radareorg/radare2/commit/00e8fd12533a6a6753dcc62e95e0eae8a74dc0e4

linked to: https://github.com/rizinorg/rizin/issues/747

### CODE REVIEW

File: librz/include/rz_sign.h [hunk 1]
```diff
@@ -27,11 +27,12 @@ typedef enum {
...
 	RZ_SIGN_OFFSET = 'o', // addr
 	RZ_SIGN_NAME = 'n', // real name
-	RZ_SIGN_REFS = 'r', // xrefs from
-	RZ_SIGN_XREFS = 'x', // xrefs to
+	RZ_SIGN_XREFS_FROM = 'r', // xrefs from
+	RZ_SIGN_XREFS_TO = 'x', // xrefs to
 	RZ_SIGN_VARS = 'v', // variables
 	RZ_SIGN_TYPES = 't', // types
 	RZ_SIGN_BBHASH = 'h', // basic block hash
+	RZ_SIGN_END = '\x00', // used for sentenal value
```
Comment (XVilka) [inline librz/include/rz_sign.h]: Sentinel

File: librz/analysis/sign.c [hunk 2]
```diff
@@ -1753,140 +1733,59 @@ static void print_function_args_json(RzAnalysis *a, PJ *pj, char *arg_type) {
...
+		if (fmt == 'j') {
 			pj_s(pj, ref);
 		} else {
 			a->cb_printf("%s", ref);
 		}
 		i++;
 	}
 
-	if (format == 'j') {
+	if (fmt == 'j') {
```
Comment (XVilka) [inline librz/analysis/sign.c]: Need to be converted to the `RzOutputMode` enum after

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by wargio** [comment]
`[XX] db/archos/linux-x64/dbg_oo ood check for ptrace errors` is a false positive. 100% not related to this PR

**Review by ret2libc — APPROVED** [review]

**Comment by ret2libc** [comment]
CI was red even for windows. Let's see.

**Comment by wargio** [comment]
I have no clue why this is happening.

**Comment by wargio** [comment]
i will convert this to draft for now.

**Pull request closed without merging**